Modify content to align with the present (#374)
giantswarm/docs: src/content/guides/limiting-pod-communication-with-network-policies/index.md

+++
title = "Limiting Pod Communication with Network Policies"
description = "Guide on how to limit Pod communication using Network Policies"

> date = "2019-11-12"

Reviewer: That's in the future ;-)

> date = "2019-11-12"

Reviewer: ```suggestion
date = "2019-11-07"
```

A date in the future will make this page invisible until we build again on or after that date.

You can limit communication to Pods using the Network Policy API of Kubernetes.

> The Network policy functionality is implemented by a network provider thanks to the CNI plugin. Nowadays there are some provides compatible (Calico, Cilium, Kube-router, ...). Giant Swarm uses Calico as a provider so out of the box users can make use of these policies.

Reviewer 2: ```suggestion
The Kubernetes Network Policy functionality is implemented by different network providers , like Calico, Cilium, Kube-router, etc. Most of these providers have some added functionalties that go above the main Kubernetes Network Policy API. Giant Swarm deploys Calico as a provider so users can make use of its functionality out of the box.
```

Note that the namespace needs to exist before you apply the NetworkPolicy to it.

The default policy shown above will limit ingress and egress traffic in the namespace applied. You can restrict only for `egress` or `ingress` too. 

## Applications 

### Allowing specific system pod to talk with your pod

As we mentioned before we harden the clusters restricting the communication with kube system and giantswarm pods. In case you need to allow that communication with a running pod in one of those namespaces you have to explicitly declare. For example:

```yaml
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  name: ksm-can-be-accessed-by-my-app
  namespace: kube-system
spec:
  podSelector:
    matchLabels:
      app: kube-state-metrics
  ingress:
    - from:
      - podSelector:
          matchLabels:
            app: my-app-that-needs-access-to-ksm
      ports:
        - protocol: TCP
          port: 10301
```

To make it more visual, this how it will look at the communication between namespaces.

> (PLACEHOLDER FOR IMAGE)

Author: working in the diragram

> ![](/img/giant-swarm-network-policies-diagram.png)

Reviewer: Can you please try placing this in the same folder as `index.md` and using only the file name as a reference? In our current HUGO version that should work. (Please check locally in preview)

### Allowing specific pod to pod access

> In the following example, we allow traffic to Pods labelled `role: backend` from Pods with the `role: frontend` label and only on TCP port 6379.

Reviewer 3: ```suggestion
In the following example, we allow traffic to Pods labeled `role: backend` from Pods with the `role: frontend` label and only on TCP port 6379.
```

Both spellings are correct but labeled is preferred in 🇺🇸 which is what we aim for. https://www.grammarly.com/blog/labeled-labelled/

            name: freeforall
```

> Note that the namespace you apply this policy needs to carry a label `name:` similar to the actual name key in its metadata:

Reviewer 3: ```suggestion
Note that the namespace you apply this policy to needs to carry a label `name:` similar to the actual name key in its metadata:
```
